<a href="https://colab.research.google.com/github/catalinakeu/voila/blob/main/CatalinaUmaFinalPython_Take_Home_Exam.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Introduction

Human immunodeficiency virus (HIV) is a viral infection attacking the body’s immune system. With the progression of the disease, an acquired immunodeficiency syndrome (AIDS), the most advanced stage of the disease, may occur. HIV virus is spread via the body fluids such as blood, breast milk, semen and vaginal fluids. With no approved vaccines against HIV, the only current prevention method is causciousness when it comes to blood transfussion, breastfeedeng and intercourse. In this report we will examine how the status of a country affect HIV/AIDS as evaluated through schooling levels. Our hypothesis is that the developed coutris will have much lower HIV/AIDS levels than the developing ones and that in countries where an increase in Schooling leves is present, the HIV/AIDS levels will drop.

**Subquestions:**

We divided the work based on different tasks to answer our research and therefore didn't work according to sperate subquestions. We divided the work between statistics with the comparison graph between countries and the interactive graph creation.

## Data preparation, cleaning and exploration

In [2]:
import pandas as pd
factors = pd.read_csv("https://raw.githubusercontent.com/NHameleers/dtz2025-datasets/master/CountryHealthFactors.csv")
factors.head()

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0            263.0   
1  Afghanistan  2014  Developing              59.9            271.0   
2  Afghanistan  2013  Developing              59.9            268.0   
3  Afghanistan  2012  Developing              59.5            272.0   
4  Afghanistan  2011  Developing              59.2            275.0   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624         65.0      1154  ...   
1             64     0.01               73.523582         62.0       492  ...   
2             66     0.01               73.219243         64.0       430  ...   
3             69     0.01               78.184215         67.0      2787  ...   
4             71     0.01                7.097109         68.0      3013  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0    6.0               8.16         65.0        0.1  584.259210  33736494.0   
1   58.0               8.18         62.0        0.1  612.696514    327582.0   
2   62.0               8.13         64.0        0.1  631.744976  31731688.0   
3   67.0               8.52         67.0        0.1  669.959000   3696958.0   
4   68.0               7.87         68.0        0.1   63.537231   2978599.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  

[5 rows x 22 columns]

The first step we took in the data analysis was deleting the coulmns that were irrelevant to our research question, however, we ran into an issue of some coulmn names being 'non-existing'. We decided to use the .rename method to get rid of the extra spaces in some coulmn names.

In [3]:
factors = factors.rename(columns=str.strip)

In [4]:
factors_used = factors.drop(['Life expectancy', 'Adult Mortality', 'infant deaths', 'Alcohol', 'percentage expenditure', 'Hepatitis B', 'Measles', 'BMI', 'under-five deaths', 'Polio', 'Total expenditure', 'Diphtheria', 'GDP', 'Population', 'thinness  1-19 years', 'thinness 5-9 years', 'Income composition of resources', ], axis=1)
factors_used.head()

Country  Year      Status  HIV/AIDS  Schooling
0  Afghanistan  2015  Developing       0.1       10.1
1  Afghanistan  2014  Developing       0.1       10.0
2  Afghanistan  2013  Developing       0.1        9.9
3  Afghanistan  2012  Developing       0.1        9.8
4  Afghanistan  2011  Developing       0.1        9.5

After removing the coumns irrelevant to our research question, we checked for the missing values in the remaining colums and deleted all the rows where any information was missing.

In [5]:
factors_used.isnull().sum()

Country        0
Year           0
Status         0
HIV/AIDS       0
Schooling    163
dtype: int64

In [6]:
factors_cleaned = factors_used.dropna(subset=['Schooling'])

In [7]:
factors_cleaned.isnull().sum()

Country      0
Year         0
Status       0
HIV/AIDS     0
Schooling    0
dtype: int64

Since the variable Status is a string variable we decided to aslo check whether there were any tipfelers that could affect our results later on.

In [8]:
factors_cleaned['Status'].value_counts()

Developing    2311
Developed      464
Name: Status, dtype: int64

Since, only two values, Developing and Developed, were found, we could proceed without having to correct anything.

Once the missing values of Schooling variable have been removed and the Status variable was checked for typos, we decided to check whether there is a difference in means of Schooling levels between Developing and Developed countries in order to check for validity of our approach to the research question.

In [9]:
factors_cleaned.groupby('Status')['Schooling'].mean()

Status
Developed     15.845474
Developing    11.219256
Name: Schooling, dtype: float64

To get more statictically relevat data, the propper statistical tests should be performed, however, for now we concluded that the difference is big enough and decided to proceed with the proposed research question.

---



To answer our research question, we decided to limit our analysis to the three developed countries with the highest mean Schooling levels and the three developing countries with the lowest mean Schooling level higher than 0.

In [10]:
developed_countries = factors_cleaned.loc[factors.Status == "Developed", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]


In [11]:
developed_countries.groupby('Country')['Schooling'].mean()
mean_schooling_by_country_developed = developed_countries.groupby('Country')['Schooling'].mean()
mean_schooling_by_country_developed.sort_values(ascending=False)

Country
Australia      20.03750
New Zealand    18.86875
Iceland        18.15625
Ireland        17.65625
Norway         17.46875
Denmark        17.19375
Netherlands    17.05625
Belgium        16.78750
Germany        16.60000
Slovenia       16.47500
Spain          16.35625
Lithuania      16.10000
Portugal       15.93750
Italy          15.93125
Sweden         15.86875
Latvia         15.56875
Switzerland    15.39375
Austria        15.38750
Poland         15.25000
Hungary        15.11875
Japan          14.97500
Slovakia       14.35000
Malta          14.30625
Singapore      13.98125
Romania        13.92500
Croatia        13.88750
Bulgaria       13.72500
Luxembourg     13.63750
Cyprus         13.51875
Name: Schooling, dtype: float64

In [12]:
developing_countries = factors_cleaned.loc[factors.Status == "Developing", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]

In [13]:
developing_countries.groupby('Country')['Schooling'].mean()
mean_schooling_by_country_developing = developing_countries.groupby('Country')['Schooling'].mean()
mean_schooling_by_country_developing.sort_values(ascending=True)

Country
Tuvalu               0.00000
Marshall Islands     0.00000
South Sudan          1.53125
Niger                4.01875
Djibouti             4.75000
                      ...   
France              15.90000
Greece              15.93750
Estonia             15.93750
Argentina           16.50625
Finland             17.29375
Name: Schooling, Length: 151, dtype: float64

After the analysis showed above, the selected developed countires were Australia, New Zealand and Iceland, and the selected developing countries were South Sudan, Niger and Djibouti.

The next step was to exclude all of the countries that were not selected for this research question.

In [14]:
australia = factors_cleaned.loc[factors_cleaned.Country == "Australia", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]

In [15]:
new_zealand = factors_cleaned.loc[factors_cleaned.Country == "New Zealand", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]


In [16]:
iceland = factors_cleaned.loc[factors_cleaned.Country == "Iceland", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]

In [17]:
djibouti = factors_cleaned.loc[factors_cleaned.Country == "Djibouti", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]

In [18]:
niger = factors_cleaned.loc[factors_cleaned.Country == "Niger", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]

In [19]:
south_sudan = factors_cleaned.loc[factors_cleaned.Country == "South Sudan", ['Year', 'Status', 'Country','HIV/AIDS', 'Schooling']]

In [20]:
final = pd.concat([australia,new_zealand,iceland,djibouti,niger,south_sudan])

In [21]:
final.head()

Year     Status    Country  HIV/AIDS  Schooling
112  2015  Developed  Australia       0.1       20.4
113  2014  Developed  Australia       0.1       20.4
114  2013  Developed  Australia       0.1       20.3
115  2012  Developed  Australia       0.1       20.1
116  2011  Developed  Australia       0.1       19.8

Lastly, a descriptive statistics table of the filtered data was created to check for any potential outliers within the data set.

In [22]:
final.describe()

Year   HIV/AIDS  Schooling
count    96.000000  96.000000  96.000000
mean   2007.500000   1.406250  11.227083
std       4.633971   1.589559   8.009974
min    2000.000000   0.100000   0.000000
25%    2003.750000   0.100000   3.950000
50%    2007.500000   0.250000  11.550000
75%    2011.250000   3.125000  19.100000
max    2015.000000   4.300000  20.700000

The minimum value for the Year variable was 2000 and the maximum value was 2015, we conclude that there were no typos and outliers in this variable. Same holds true for the variable HIV/AIDS where the minumum and maximum value were 0.1 and 4.3 respectfully. Lastly, the Schooling variable values ranged between 0.0 and 20,7 which also seems in line with all the previously created tables and shows no outliers or typos

## Data description

Below, six tables, showing the descriptive statistics of each country separately, are presented.

In [23]:
australia.describe()

Year  HIV/AIDS  Schooling
count    16.000000      16.0  16.000000
mean   2007.500000       0.1  20.037500
std       4.760952       0.0   0.563176
min    2000.000000       0.1  19.000000
25%    2003.750000       0.1  19.725000
50%    2007.500000       0.1  20.300000
75%    2011.250000       0.1  20.400000
max    2015.000000       0.1  20.700000

The descriptives statisctis table of the data collected in Australia shows that the data has been collected every year from 2000 to 2015. In said years the HIV/AIDS level has been constant at 0.1 (mean = 0.1, std = 0.0). The schooling level varied between 19.0 and 20.7, remaining at a relatively constant level as well.

In [24]:
new_zealand.describe()

Year  HIV/AIDS  Schooling
count    16.000000      16.0  16.000000
mean   2007.500000       0.1  18.868750
std       4.760952       0.0   0.885979
min    2000.000000       0.1  17.200000
25%    2003.750000       0.1  18.350000
50%    2007.500000       0.1  19.200000
75%    2011.250000       0.1  19.350000
max    2015.000000       0.1  20.300000

The descriptives statisctis table of the data collected in New Zealand shows that the data has been collected every year from 2000 to 2015. In said years the HIV/AIDS level has been constant at 0.1 (mean = 0.1, std = 0.0). The schooling level varied between 17.2 and 20.3, showing more variation than australia, but still maintaining a relatively constant level.

In [25]:
iceland.describe()

Year  HIV/AIDS  Schooling
count    16.000000      16.0  16.000000
mean   2007.500000       0.1  18.156250
std       4.760952       0.0   0.684075
min    2000.000000       0.1  16.800000
25%    2003.750000       0.1  17.825000
50%    2007.500000       0.1  18.250000
75%    2011.250000       0.1  18.625000
max    2015.000000       0.1  19.000000

The descriptives statisctis table of the data collected in Iceland shows that the data has been collected every year from 2000 to 2015. In said years the HIV/AIDS level has been constant at 0.1 (mean = 0.1, std = 0.0). The schooling level varied between 16.8 and 19, once again remaining at a relatively constant level.

In [26]:
djibouti.describe()

Year   HIV/AIDS  Schooling
count    16.000000  16.000000  16.000000
mean   2007.500000   3.281250   4.750000
std       4.760952   0.863496   1.272792
min    2000.000000   1.900000   2.900000
25%    2003.750000   2.550000   3.650000
50%    2007.500000   3.400000   4.750000
75%    2011.250000   4.025000   6.000000
max    2015.000000   4.300000   6.300000

The descriptives statisctis table of the data collected in Dijbouti shows that the data has been collected every year from 2000 to 2015. In said years the HIV/AIDS level varied between 1.9 and 4.3 witha std of 0.86. The schooling level varied between 2.9 and 6.3.

In [27]:
niger.describe()

Year   HIV/AIDS  Schooling
count    16.000000  16.000000  16.000000
mean   2007.500000   1.162500   4.018750
std       4.760952   0.454423   0.949539
min    2000.000000   0.400000   2.800000
25%    2003.750000   0.700000   3.075000
50%    2007.500000   1.350000   3.900000
75%    2011.250000   1.600000   4.875000
max    2015.000000   1.600000   5.400000

The descriptives statisctis table of the data collected in Niger shows that the data has been collected every year from 2000 to 2015. In said years the HIV/AIDS level varied between 0.4 and 1.6 witha std of 0.45. The schooling level varied between 2.8 and 5.4.

In [28]:
south_sudan.describe()

Year   HIV/AIDS  Schooling
count    16.000000  16.000000  16.000000
mean   2007.500000   3.693750   1.531250
std       4.760952   0.441918   2.345696
min    2000.000000   2.700000   0.000000
25%    2003.750000   3.475000   0.000000
50%    2007.500000   3.800000   0.000000
75%    2011.250000   4.025000   4.900000
max    2015.000000   4.200000   4.900000

The descriptives statisctis table of the data collected in South Sudan shows that the data has been collected every year from 2000 to 2015. In said years the HIV/AIDS level varied between 2.7 and 4.2 witha std of 0.44. The schooling level varied between 0.0 and 4.9.

## Data visualisation

In order to visually present our data two graphs were created, one showing the change in HIV/AIDS levels over the years in different countries, and the other one showing Schooling levels over the years in different countries. For an easier comparison of the graphs they were arranged horizontally.

In [29]:
import altair as alt

In [30]:
HIV_graph = alt.Chart(final).mark_line().encode(
    x='Year:N',
    y='HIV/AIDS',
    color='Country'
).properties(title='HIV/AID versus Year per Country')
Schooling_graph = alt.Chart(final).mark_line().encode(
    x='Year:N',
    y='Schooling',
    color='Country'
).properties(title='Schooling versus Year per Country')
HIV_graph | Schooling_graph


alt.HConcatChart(...)

All three developed countries, Australia, New Zealand and Iceland, show a steady, low level of HIV/AIDS throughout the years (0.1) as well as a slightly alterating, but still relatively high level of Schooling. Contrary, two developing countries, Djibouti and Niger, show a decrease in HIV/AIDS levels thoroughout the years as well as an increasing level of Schooling. In South Sudan the level of Schooling up until 2010 was 0.0, during which time the HIV/AIDS levels increased. With an increase of Schooling, the HIV/AIDS levels started dropping but were in 2015 still higher than in 2000.

Additionally, a graph showing the severity of HIV/AIDS in a population in a certain country associated with the level of schooling in said country is shown

In [31]:
brush = alt.selection(type='interval')

points = alt.Chart(final).mark_circle().encode(
    x='Schooling',
    y='Country',
    color=alt.condition(brush, 'HIV/AIDS:Q', alt.value('blue')),
    size='HIV/AIDS'
).add_selection(
    brush
)
points

alt.Chart(...)

An interactive component was added to the graph above where readers can use the slider to select the year and on a graph a display is shown of the HIV/AID level in each country with the responding level of schooling in said year. This interactive visualization allowes for the tracking of HIV/AIDS schooling relationship in different countries through time

In [49]:
import pandas as pd
import altair as alt
import ipywidgets as widgets
from ipywidgets import interact

final = pd.DataFrame(final)

year_slider = widgets.IntSlider(min=2000, max=2015, step=1, value=2010)
widgets.HBox([year_slider])

brush = alt.selection(type='interval')

def interactive_chart(selected_year):
    interactive_data = final[final['Year'] == selected_year]

    chart = alt.Chart(interactive_data).mark_circle(size=100).encode(
        x='Schooling',
        y='Country',
        color=alt.condition(brush, 'HIV/AIDS:Q', alt.value('blue')),
        size=alt.Size('HIV/AIDS:Q', scale=alt.Scale(range=[55, 150]))
    ).properties(
        title=f'Scatter Plot for Year {selected_year}'
    ).add_params(
        brush
    )

    display(chart)

interact(interactive_chart, selected_year=year_slider);

interactive(children=(IntSlider(value=2010, description='selected_year', max=2015, min=2000), Output()), _dom_…

## Conclusion

From the above presented data, a conclusion can be drawn that in the developed countries the HIV/AIDS levels are minimal, while in the developing they are higher but show a negative correlation with the Schooling levels, confirming both our hypotheses. It is interesting to mention that the similar difference in schooling levels throughout the years shows no effect on HIV/AIDS in developed countries and a noticable change in developing countries.  To further check these findings and get more insight into the exact relationship between the two factors, additional researches should be conducted.